libs/SmartSync/src/com/salesforce/androidsdk/smartsync/model/Layout.java

```diff
@@ -0,0 +1,398 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.smartsync.model;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the layout of a Salesforce object.
+ *
+ * @author bhariharan
+ * @see <a href="https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_responses_record_layout.htm">https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_responses_record_layout.htm</a>
+ */
+public class Layout {
+
+    private static final String ID = "id";
+    private static final String LAYOUT_TYPE = "layoutType";
+    private static final String MODE = "mode";
+    private static final String SECTIONS = "sections";
+
+    private String id;
+    private String layoutType;
+    private String mode;
+    private List<LayoutSection> sections;
+    private JSONObject rawData;
+
+    /**
+     * Creates an instance of this class from its JSON representation.
+     *
+     * @param object JSON object.
+     * @return Instance of this class.
+     */
+    public static Layout fromJSON(JSONObject object) {
+        Layout layout = null;
+        if (object != null) {
+            layout = new Layout();
+            layout.rawData = object;
+            layout.id = object.optString(ID);
+            layout.layoutType = object.optString(LAYOUT_TYPE);
+            layout.mode = object.optString(MODE);
+            final JSONArray sections = object.optJSONArray(SECTIONS);
+            if (sections != null) {
+                layout.sections = new ArrayList<>();
+                for (int i = 0; i < sections.length(); i++) {
+                    final JSONObject section = sections.optJSONObject(i);
+                    if (section != null) {
+                        layout.sections.add(LayoutSection.fromJSON(section));
+                    }
+                }
+            }
+        }
+        return layout;
+    }
+
+    /**
+     * Returns the ID of this layout.
+     *
+     * @return ID of this layout.
+     */
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Returns the layout type of this layout.
+     *
+     * @return Layout type of this layout.
+     */
+    public String getLayoutType() {
+        return layoutType;
+    }
+
+    /**
+     * Returns the mode of this layout.
+     *
+     * @return Mode of this layout.
+     */
+    public String getMode() {
```

Comment:
Should that return an enum type?

Reply:
Nope - this is not our cache mode. This is the mode in the payload of the layout response itself, could be `Create`, `Update`, etc. It represents record display mode I think.

Reply:
If there is a fixed set of possible values, it would be better represented with an enum.

Reply:
The UI API doc does not call out whether it's a fixed set of values or not. I'd prefer to mirror the API response data model per UI API doc and avoid mutation on the client. If the server doc is updated to specify a set of possible values (`enum`), we can update this to `enum` at that time.